### PR TITLE
[DOCS] Update GX Cloud Documentation to Reflect New Data Asset Workflow

### PR DESCRIPTION
**Pull request by kwcanuck**
The GX Cloud UI now includes code samples for connecting GX Cloud to Snowflake and PostgreSQL. This change necessitated a rewrite of some of the existing procedures. This PR implements the necessary edits. 

## Definition of done
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [X] Appropriate tests and docs have been updated

### CODE REVIEW

**Comment by annabarr** [comment]
@kwcanuck Can you share more about your decision to remove the code snippet from the docs in favor of the in-product snippet, rather than have them live in both places?

@annabarr, the code snippets are now available in the UI and are part of the workflow. If the user follows the written instructions, they will be able to copy the code as it is right in front of them. Adding the code snippets to the documentation is now redundant and unnecessarily lengthens the procedures. In addition, a user can choose not to copy the code; therefore, the code snippets might be unnecessary and more of a nuisance than a help.

**Comment by annabarr** [comment]
> @kwcanuck Can you share more about your decision to remove the code snippet from the docs in favor of the in-product snippet, rather than have them live in both places?
> 
> @annabarr, the code snippets are now available in the UI and are part of the workflow. If the user follows the written instructions, they will be able to copy the code as it is right in front of them. Adding the code snippets to the documentation is now redundant and unnecessarily lengthens the procedures. In addition, a user can choose not to copy the code; therefore, the code snippets might be unnecessary and more of a nuisance than a help.

@kwcanuck I wonder if we are making it more difficult for users to complete the "Prepare your environment" step, to have the application, the docs, and the data source open at the same time, rather than the application and the data source, or the docs and the data source.

@annabarr, in my opinion, adding the code snippet to the UI isn't helpful. Not all users will want to configure a production deployment. Also, users are greeted with a huge code snippet that could be intimidating. We're lengthening the process for adding a Data Asset, and now there are three dialogs a user needs to navigate. No matter how the docs are written, we're asking a LOT of our users. They have no choice but to have the UI, the Data Source, and the docs open. This isn't unique to GX. The documentation as it's currently written is accurate, and I am confident users will be successful if they follow the instructions.   Let's meet and chat if you remain unconvinced that my approach is correct. It's easier to explain it in a conversation.

**Review by annabarr — APPROVED** [review]

File: docs/docusaurus/docs/cloud/connect/connect_postgresql.md
Comment (Rachel-Reverie) [inline docs/docusaurus/docs/cloud/connect/connect_postgresql.md]: This seems a little unclear.  You aren't so much entering a name for the table you're creating as you're entering the name of the table the Data Asset connects to.

File: docs/docusaurus/docs/cloud/connect/connect_postgresql.md
Comment (annabarr) [inline docs/docusaurus/docs/cloud/connect/connect_postgresql.md]: This is technically the name for the Data Source. The data asset names are defined later

File: docs/docusaurus/docs/cloud/connect/connect_postgresql.md
Comment (Rachel-Reverie) [inline docs/docusaurus/docs/cloud/connect/connect_postgresql.md]: I think this is slightly inaccurate.  When query asset is selected, there is no "the table" specified by the user.  Instead, since the query could theoretically join multiple tables, it should be something like:
```suggestion
    - **Query**: When **Query Asset** is selected, enter the query that you want to run on the database.
```

File: docs/docusaurus/docs/cloud/connect/connect_snowflake.md
Comment (annabarr) [inline docs/docusaurus/docs/cloud/connect/connect_snowflake.md:32]: There is another step to select Snowflake or Postgres in between 1 and 2.

File: docs/docusaurus/docs/cloud/connect/connect_postgresql.md
Comment (Rachel-Reverie) [inline docs/docusaurus/docs/cloud/connect/connect_postgresql.md]: Maybe:
```suggestion
    - **Table name**: When **Table Asset** is selected, enter the name of the table you're connecting to with the Data Asset.
```

File: docs/docusaurus/docs/cloud/connect/connect_snowflake.md
Comment (Rachel-Reverie) [inline docs/docusaurus/docs/cloud/connect/connect_snowflake.md:66]: Here we're telling people to click things, but not telling them why or what to do with what they get back.  Perhaps:
```suggestion
    - **Database**: Enter the name of the Snowflake database where the data you want to validate is stored. To find the names of available databases in Snowsight, click **Data** > **Databases**. In the Snowflake Classic Console, find the available database names by clicking **Databases**.
```

File: docs/docusaurus/docs/cloud/connect/connect_snowflake.md
Comment (Rachel-Reverie) [inline docs/docusaurus/docs/cloud/connect/connect_snowflake.md:70]: As with databases, this doesn't explain that they're clicking around in Snowsight/Snowflake Classic Console to find the names that can be entered as valid Warehouse values.

File: docs/docusaurus/docs/cloud/connect/connect_snowflake.md
Comment (Rachel-Reverie) [inline docs/docusaurus/docs/cloud/connect/connect_snowflake.md]: As with the PostgreSQL option, I think this is a little inaccurate to what the user is actually entering and what is being done with that value.
```suggestion
    - **Table name**: When **Table Asset** is selected, enter the name of the table you're connecting to with the Data Asset.
```

File: docs/docusaurus/docs/cloud/connect/connect_snowflake.md
Comment (Rachel-Reverie) [inline docs/docusaurus/docs/cloud/connect/connect_snowflake.md]: Same as in the PostgreSQL guide:
```suggestion
    - **Query**: When **Query Asset** is selected, enter the query that you want to run on the database. 
```

File: docs/docusaurus/docs/cloud/data_assets/manage_data_assets.md
Comment (Rachel-Reverie) [inline docs/docusaurus/docs/cloud/data_assets/manage_data_assets.md:52]: Copy what code?  Isn't this one of the processes that is entirely in the pop-out drawer in GX Cloud?

File: docs/docusaurus/docs/cloud/data_assets/manage_data_assets.md
Comment (Rachel-Reverie) [inline docs/docusaurus/docs/cloud/data_assets/manage_data_assets.md:131]: I don't think this is supposed to be a "meaningful name for the Data Asset."  Shouldn't this be a meaningful name for the Data Source, and the Data Asset gets named later?

**Review by Rachel-Reverie — APPROVED** [review]
I added some comments and suggestions where I thought things could be clearer.  And for one spot (the **Data Source name** field) I think had incorrect instructions.  Since these changes cross versions, make sure that any suggestions you implement get applied to all instances of that text.  

Otherwise, LGTM.  I'll approve so you aren't blocked on me after you make any changes per the suggestions.

File: docs/docusaurus/docs/cloud/connect/connect_postgresql.md
Comment (kwcanuck) [reply docs/docusaurus/docs/cloud/connect/connect_postgresql.md]: @Rachel-Reverie, I attempted to clarify the text.

File: docs/docusaurus/docs/cloud/connect/connect_postgresql.md
Comment (kwcanuck) [reply docs/docusaurus/docs/cloud/connect/connect_postgresql.md]: @Rachel-Reverie, I attempted to clarify the text.

File: docs/docusaurus/docs/cloud/connect/connect_postgresql.md
Comment (kwcanuck) [reply docs/docusaurus/docs/cloud/connect/connect_postgresql.md]: @annabarr I have updated the text.

File: docs/docusaurus/docs/cloud/connect/connect_snowflake.md
Comment (kwcanuck) [reply docs/docusaurus/docs/cloud/connect/connect_snowflake.md:32]: @annabarr good catch! I was using the Figma mockup and missed this. The documentation has been updated.

File: docs/docusaurus/docs/cloud/connect/connect_snowflake.md
Comment (kwcanuck) [reply docs/docusaurus/docs/cloud/connect/connect_snowflake.md:66]: @Rachel-Reverie, if you're in the application, it is clear what you're doing. Also, the user is instructed to "enter the name of the Snowflake database", so it's safe to assume that they know what the instructions are for.

File: docs/docusaurus/docs/cloud/connect/connect_snowflake.md
Comment (kwcanuck) [reply docs/docusaurus/docs/cloud/connect/connect_snowflake.md:70]: @Rachel-Reverie see my previous comment.

File: docs/docusaurus/docs/cloud/connect/connect_snowflake.md
Comment (kwcanuck) [reply docs/docusaurus/docs/cloud/connect/connect_snowflake.md]: @Rachel-Reverie the text has been updated.

File: docs/docusaurus/docs/cloud/connect/connect_snowflake.md
Comment (kwcanuck) [reply docs/docusaurus/docs/cloud/connect/connect_snowflake.md]: @Rachel-Reverie the text has been updated.

File: docs/docusaurus/docs/cloud/data_assets/manage_data_assets.md
Comment (kwcanuck) [reply docs/docusaurus/docs/cloud/data_assets/manage_data_assets.md:52]: @Rachel-Reverie, the user is instructed to "copy the code in the code pane." So, they know where it is. Prior to this instruction, they are instructed "in GX Cloud, click **Data Assets** > **New Data Asset** > **Snowflake**" or "in GX Cloud, click **Data Assets** > **New Data Asset** > **PostgreSQL**. Again, if you follow the procedure, you'll see what is being referenced and know what to do. There is very little risk a user won't know what to do or where to do it if they follow the instructions. For your reference, here is the updated UI:

image

File: docs/docusaurus/docs/cloud/data_assets/manage_data_assets.md
Comment (kwcanuck) [reply docs/docusaurus/docs/cloud/data_assets/manage_data_assets.md:131]: @Rachel-Reverie, this issue has been addressed. The UI has been changing frequently, so this was missed.

**Pull request merged**
